### ISSUE TITLE
qa: Failure in feature_txindex_compatibility.py on master

### ISSUE THREAD

**Issue by hebasto** [bitcoin/bitcoin#23514]
https://api.cirrus-ci.com/v1/task/6143696841211904/logs/ci.log:
```
204/225 - [1mfeature_txindex_compatibility.py[0m failed, Duration: 1 s

[1mstdout:
[0m2021-11-15T10:09:01.666000Z TestFramework (INFO): Initializing test directory /tmp/cirrus-build/ci/scratch/test_runner/test_runner_₿_🏃_20211115_095356/feature_txindex_compatibility_18
2021-11-15T10:09:02.259000Z TestFramework (ERROR): Assertion failed
Traceback (most recent call last):
  File "/tmp/cirrus-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/test_framework/test_framework.py", line 132, in main
    self.run_test()
  File "/tmp/cirrus-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/feature_txindex_compatibility.py", line 47, in run_test
    mini_wallet.send_self_transfer(from_node=self.nodes[1], utxo_to_spend=spend_utxo)
  File "/tmp/cirrus-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/test_framework/wallet.py", line 150, in send_self_transfer
    tx = self.create_self_transfer(**kwargs)
  File "/tmp/cirrus-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/test_framework/wallet.py", line 202, in create_self_transfer
    assert_equal(mempool_valid, tx_info['allowed'])
  File "/tmp/cirrus-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/test_framework/util.py", line 50, in assert_equal
    raise AssertionError("not(%s)" % " == ".join(str(arg) for arg in (thing1, thing2) + args))
AssertionError: not(True == False)
2021-11-15T10:09:02.314000Z TestFramework (INFO): Stopping nodes
2021-11-15T10:09:02.571000Z TestFramework (WARNING): Not cleaning up dir /tmp/cirrus-build/ci/scratch/test_runner/test_runner_₿_🏃_20211115_095356/feature_txindex_compatibility_18
2021-11-15T10:09:02.571000Z TestFramework (ERROR): Test failed. Test logging available at /tmp/cirrus-build/ci/scratch/test_runner/test_runner_₿_🏃_20211115_095356/feature_txindex_compatibility_18/test_framework.log
2021-11-15T10:09:02.572000Z TestFramework (ERROR): 
2021-11-15T10:09:02.572000Z TestFramework (ERROR): Hint: Call /tmp/cirrus-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/combine_logs.py '/tmp/cirrus-build/ci/scratch/test_runner/test_runner_₿_🏃_20211115_095356/feature_txindex_compatibility_18' to consolidate all logs
2021-11-15T10:09:02.572000Z TestFramework (ERROR): 
2021-11-15T10:09:02.572000Z TestFramework (ERROR): If this failure happened unexpectedly or intermittently, please file a bug and provide a link or upload of the combined log.
2021-11-15T10:09:02.572000Z TestFramework (ERROR): https://github.com/bitcoin/bitcoin/issues
2021-11-15T10:09:02.573000Z TestFramework (ERROR): 
```

**Comment by hebasto** [comment]
Also on master (41a1b5f58ca59d2177dcadf834efd187fa3fba52) the same failure is observed.

**Comment by fanquake** [comment]
There is a PR open to fix this: #23515.
